server/services/offender/responses/transaction.js

```diff
@@ -0,0 +1,56 @@
+const { formatBalanceOrDefault } = require('../../../utils/string');
+const {
+  placeholders: { DEFAULT },
+  dateFormats: { TRANSACTION_FORMAT },
+} = require('../../../utils/enums');
+const { formatDateOrDefault } = require('../../../utils/date');
+
+class Transaction {
+  constructor(options = {}) {
+    const {
+      entryDate,
+      penceAmount,
+      postingType,
+      currentBalance,
+      entryDescription,
+      agencyId,
+      currency,
+    } = options;
+    this.paymentDate = entryDate;
+    this.postingType = postingType;
+    this.amount = penceAmount / 100;
+    this.currency = currency;
+    this.balance = currentBalance / 100;
```

Comment:
Could we store currency amounts as pence until the last possible moment, and do decimal formatting in the presentation layer? Casting early runs the risk of dealing with rounding errors, so we could trip ourselves up by casting early if we used these values for any arithmetic in future.

Reply:
100% 👍 Updated to retain the pence amount until we're ready to present the data in the template